src/framework/enhancement/js/ProgressiveEnhancement.js

```diff
@@ -29,116 +26,15 @@ var fluid_2_0 = fluid_2_0 || {};
         return !!window.FormData;
     };
 
-    /*
-     * An object to hold the results of the progressive enhancement checks.
-     * Keys represent the key into the static environment
-     * Values represent the result of the check
-     */
-    // unsupported, NON-API value
-    fluid.enhance.checked = {};
-
-    /*
-     * The segment separator used by fluid.enhance.typeToKey
-     */
-    // unsupported, NON-API value
-    fluid.enhance.sep = "--";
-
-    /*
-     * Converts a type tag name to one that is safe to use as a key in an object, by replacing all of the "."
-     * with the separator specified at fluid.enhance.sep
-     */
-    // unsupported, NON-API function
-    fluid.enhance.typeToKey = function (typeName) {
-        return typeName.replace(/[.]/gi, fluid.enhance.sep);
-    };
-
-    /*
-     * Takes an object of key/value pairs where the key will be the key in the static environment and the value is a function or function name to run.
-     * {staticEnvKey: "progressiveCheckFunc"}
-     * Note that the function will not be run if its result is already recorded.
-     */
-    fluid.enhance.check = function (stuffToCheck) {
-        fluid.each(stuffToCheck, function (val, key) {
-            var staticKey = fluid.enhance.typeToKey(key);
-
-            if (fluid.enhance.checked[staticKey] === undefined) {
-                var results = typeof(val) === "boolean" ? val :
-                    (typeof(val) === "string" ? fluid.invokeGlobalFunction(val) : val());
-
-                fluid.enhance.checked[staticKey] = !!results;
-
-                if (results) {
-                    fluid.staticEnvironment[staticKey] = fluid.typeTag(key);
-                }
-            }
-        });
-    };
-
-    /*
-     * forgets a single item based on the typeName
-     */
-    fluid.enhance.forget = function (typeName) {
-        var key = fluid.enhance.typeToKey(typeName);
-
-        if (fluid.enhance.checked[key] !== undefined) {
-            delete fluid.staticEnvironment[key];
-            delete fluid.enhance.checked[key];
-        }
-    };
-
-    /*
-     * forgets all of the keys added by fluid.enhance.check
-     */
-    fluid.enhance.forgetAll = function () {
-        fluid.each(fluid.enhance.checked, function (val, key) {
-            fluid.enhance.forget(key);
-        });
-    };
-
-    fluid.defaults("fluid.progressiveChecker", {
-        gradeNames: ["fluid.typeFount", "fluid.littleComponent", "autoInit", "{that}.check"],
-        checks: [], // [{"feature": "{IoC Expression}", "contextName": "context.name"}]
-        defaultContextName: undefined,
-        invokers: {
-            check: {
-                funcName: "fluid.progressiveChecker.check",
-                args: ["{that}.options.checks", "{that}.options.defaultContextName"]
-            }
+    fluid.contextAware.makeChecks({
```

Comment:
Given that we've moved checks like <code>isBrowser</code> up to ContextAwareness.js, and these checks are pretty uncommon now (all modern browsers implement XHR 2.0), should they be moved into the Uploader instead?